app/scripts/controllers/preferences.js

```diff
@@ -708,6 +729,8 @@ export default class PreferencesController {
   _subscribeToNetworkDidChange() {
     this.network.on(NETWORK_EVENTS.NETWORK_DID_CHANGE, () => {
       const { tokens, hiddenTokens } = this._getTokenRelatedStates();
+      const { type, rpcUrl } = this.network.getProviderConfig();
+      this.provider = ethers.getDefaultProvider(rpcUrl || type);
```

Comment:
I'd recommend using the `provider` we use everywhere else in the background, just to ensure this behaves as expected. I don't know if there are any differences in behavior, but just something to consider. You can see an example of how that can be passed into `ethers` in the `swaps` controller.

Reply:
I thought I'd tried this before but apparently not correctly. I am making this change now.